### PR TITLE
Remove unnecessary request pattern usages

### PR DESCRIPTION
**Pull request by anthony-murphy**
We used to require IFluidHandles to also be IFluidRouters. This was changed long ago, but it looks like some remanent were missed.

### CODE REVIEW

File: packages/runtime/container-runtime/src/summarizer.ts [hunk 1]
```diff
@@ -8,16 +8,14 @@ import { Deferred } from "@fluidframework/common-utils";
 import { ITelemetryLogger } from "@fluidframework/common-definitions";
 import { ChildLogger, IFluidErrorBase, LoggingError, wrapErrorAndLog } from "@fluidframework/telemetry-utils";
 import {
-    IRequest,
-    IResponse,
+
```
Comment (ChumpChief) [inline packages/runtime/container-runtime/src/summarizer.ts:11]:
```suggestion
```

File: packages/runtime/container-runtime/src/containerHandleContext.ts [hunk 2]
```diff
@@ -13,7 +13,6 @@ import { generateHandleContextPath } from "@fluidframework/runtime-utils";
 import { ContainerRuntime } from "./containerRuntime";
 
 export class ContainerFluidHandleContext implements IFluidHandleContext {
-    public get IFluidRouter() { return this; }
```
Comment (ChumpChief) [inline packages/runtime/container-runtime/src/containerHandleContext.ts:16]: Kind of raises an interesting question of whether we have any other places where we falsely identify as an `IFluid*` without actually implementing the interface.

**Review by ChumpChief — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerHandleContext.ts (hunk 2 above)
Comment (anthony-murphy) [reply packages/runtime/container-runtime/src/containerHandleContext.ts:16]: it's probably happens. we should probably enforce a lint rule that force typing on properties, as that would give us a compile time failure.

**Review by vladsud — APPROVED** [review]
I wanted to do the same some time ago, but never got to it :)

**Pull request merged**